platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,881 @@
+give a network a ledger id and create a way of signing messages such that the signatures can be verified by the
+ledger id.
+
+In this BLS based TSS, the network is assigned a durable BLS private/public key pair where the public key is the ledger
+id and the private key is a secret that no one knows. Each node in the network is given a number of shares  
+proportional to its signing weight. Each share is a BLS key on the same elliptic curve as the ledger key. When the
+network needs to sign a block root hash, each node uses its shares to sign the block root hash and gossips the
+signature out to the network. When a node has collected a threshold number of signatures on the same block root hash,
+it can aggregate the signatures into a ledger signature that is verifiable by the ledger id.
+
+The TSS effort has been broken down into four separate proposals: TSS-Library, TSS-Roster, TSS-Ledger-ID, and
+TSS-Block-Signing.
+
+1. The `TSS-Library` proposal contains the cryptographic primitives and algorithms needed to implement TSS.
+2. The `TSS-Roster` proposal introduces the data structure of a consensus `Roster` to replace the platform's concept of
+   an `AddressBook` and modifies the life-cycle for when the platform receives new consensus rosters.
+3. This proposal (`TSS-Ledger-ID`) depends on the first two proposals and is for the integration of a threshold
+   signature scheme into the consensus node, giving the network a ledger id that can verify ledger signatures.
+4. The `TSS-Block-Signing` proposal is everything needed to support the signing of blocks.
+
+This  `TSS-Ledger-ID` proposal covers changes to the following elements:
+
+* The process of generating the ledger key pair for new networks and existing networks.
+* The process of transferring the ability of the ledger to sign a message from one set of consensus nodes to another.
+* The new state data structures needed to store TSS Key material and the ledger id.
+* The new components needed in the framework to support creating and transferring the ledger key.
+* The modified startup process to initialize the signing capability.
+* For each node, the creation of a new Elliptic Curve (EC) key called its `tssEncryptionKey` that is used in creating
+  shares.
+
+### Goals
+
+The following capabilities are the goals of this proposal:
+
+1. `TSS Bootstrap on Existing Network` - Able to setup an existing network with a ledger public/private key pair.
+2. `TSS Bootstrap for New Network` - Able to setup a new network with a ledger public/private key pair.
+3. `Keying The Next Roster` - Able to transfer the ability to sign a message with the ledger private key from one
+   set of consensus nodes to another.
+4. Testing of the TSS signing capability with verification by an EVM smart contract using the ALT_BN128
+   [precompiles that are available in the EVM](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1829.md).
+
+### Non-Goals
+
+The following are not goals of this proposal:
```

Comment:
I think there might be an opportunity here to highlight whether this work will impact things like ISS. For example, will the introduction of a network ledger ID (as against individual RSA signatures) change the ISS detection algorithm? Though we won't immediately be at DAB Phase 3 but in the long term when we do, will the dynamic adoption of Rosters impact ISS in any way? e.g., if roster changes are happening too frequently? (less likely, but a theoretical possibility).

I don't have an answer to these, but I think it's worth highlighting here even if it's basic details, or in the worst case, we document as unknowns.

Reply:
I'm indicating here that we are not replacing the RSA key responsibilities.  ISS remains the same.   I will add in the implications section that completion of this effort opens up the possibility of replacing some of the RSA key responsibilities.